src/test/java/htsjdk/samtools/filter/IntervalFilterTest.java

```diff
@@ -0,0 +1,67 @@
+package htsjdk.samtools.filter;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecordSetBuilder;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+/**
+ * Created by farjoun on 5/27/17.
+ */
+public class IntervalFilterTest extends HtsjdkTest {
+
+    private final SAMFileHeader fileHeader;
+    private final IntervalList list;
+
+    public IntervalFilterTest() {
+        fileHeader = IntervalList.fromFile(new File("src/test/resources/htsjdk/samtools/intervallist/IntervalListchr123_empty.interval_list")).getHeader();
+        fileHeader.setSortOrder(SAMFileHeader.SortOrder.unsorted);
+
+        list = new IntervalList(fileHeader);
+
+        list.add(new Interval("1", 50, 150));   //de-facto 1:50-150 1:301-500      2:1-150 2:250-270 2:290-400
+        list.add(new Interval("1", 301, 500));
+        list.add(new Interval("2", 1, 150));
+        list.add(new Interval("2", 250, 270));
+        list.add(new Interval("2", 300, 299)); // empty, but located.
+
+    }
+
+    @Test
+    public void testReads() {
+        SAMRecordSetBuilder builder = new SAMRecordSetBuilder();
+
+        int i = 0;
+        int expected = 0;
+        //reads are of length 36
+        builder.addPair("abutting" + i++, 0, 50 - 36, 151); //both abutting
```

Comment:
I might just define a constant for something like this, ex: `50 - READ_LENGTH` which might be clearer and eliminate the need for the comment.